### PR TITLE
add make debug with address sanitizer flag enabled

### PR DESCRIPTION
**Pull request by anuchak**
add make debug with address sanitizer flag enabled

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -23,6 +23,20 @@ jobs:
       - name: test
         run: CC=gcc CXX=g++ make test
 
+  gcc-make-debug:
+    name: gcc make debug
+    needs: [ clang-formatting-check ]
```
Comment (mewim) [inline .github/workflows/ci-workflow.yml:28]: We should make this depend on `gcc-build-test` instead. If the gcc build test does not success we should not run the debug build.

File: .github/workflows/ci-workflow.yml [hunk 2]
```diff
@@ -23,6 +23,20 @@ jobs:
...
+    runs-on: kuzu-self-hosted-testing
+    steps:
+      - uses: actions/checkout@v2
+      - run: pip install -r tools/python_api/requirements_dev.txt
```
Comment (mewim) [inline .github/workflows/ci-workflow.yml:32]: For all the `pip install` on CI, we should change to `pip install --user -r`

File: .github/workflows/ci-workflow.yml (hunk 1 above)
Comment (anuchak) [reply .github/workflows/ci-workflow.yml]: yes added that

File: .github/workflows/ci-workflow.yml (hunk 2 above)
Comment (anuchak) [reply .github/workflows/ci-workflow.yml]: done

**Pull request closed without merging**